File: library/src/main/java/com/example/application/endpoints/helloworld/HelloWorldEndpoint.java
package com.example.application.endpoints.helloworld;

import com.example.application.annotations.NeverNull;
import com.vaadin.flow.server.auth.AnonymousAllowed;
import dev.hilla.Endpoint;

@Endpoint
@AnonymousAllowed
public class HelloWorldEndpoint {

    @NeverNull
    public String sayHello(@NeverNull String name) {
        if (name.isEmpty()) {
            return "Hello stranger";
        } else {
            return "Hello " + name;
        }
    }
}

File: library/src/main/java/com/example/application/endpoints/helloworld/HelloWorldEndpoint.kt
package com.example.application.endpoints.helloworld

import com.vaadin.flow.server.auth.AnonymousAllowed
import dev.hilla.Endpoint

@Endpoint
@AnonymousAllowed
class HelloWorldEndpoint {
    fun sayHello(name: String): String {
        return if (name.isEmpty()) {
            "Hello stranger from Kotlin"
        } else {
            "Hello $name from Kotlin"
        }
    }
}
